app/models/service_template_provision_request.rb

```diff
@@ -6,6 +6,8 @@ class ServiceTemplateProvisionRequest < MiqRequest
   validates_inclusion_of :request_state,  :in => %w( pending finished ) + ACTIVE_STATES, :message => "should be pending, #{ACTIVE_STATES.join(", ")} or finished"
   validate               :must_have_user
 
+  after_create :process_service_order
```

Comment:
Does after_create do it in the same transaction?  If not, perhaps it should, that way when it replicates you get them at the same time.